crates/core/src/host/wasmtime/wasm_instance_env.rs

```diff
@@ -848,7 +848,7 @@ impl WasmInstanceEnv {
                 env.call_reducer_args = None;
                 Ok(-1i32)
             } else {
-                *cursor = can_read_len;
+                *cursor += can_read_len;
```

Comment:
@Centril why did this pass all the smoketests? Can we add test coverage for this?

Reply:
afaiu, it only fails when the reducer args are past a certain size. I didn't look close enough to know what size exactly, perhaps then we can make a test which fails in this repo.